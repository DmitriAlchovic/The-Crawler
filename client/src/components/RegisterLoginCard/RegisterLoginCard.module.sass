.login-register-container
    display: flex
    flex-direction: column
    justify-content: center
    align-items: center

    .label
        font-weight: 500
        font-size: 14px
        position: relative
        display: flex
        flex-direction: column
        align-self: center 
        text-align: start

        .input
            width: 224px
            color: grey
            font-weight: 500
            font-size: 14px
            border: none
            padding: 12px
            width: inherit
            height: inherit
            height: 2.2rem

    .login-register-btn
        background-color: #3BB77E
        color: #ffff
        width: 204px
        height: 40px
        display: flex
        justify-content: center
        align-self: center
        align-items: center
        font-size: 14px
        margin: 35px 10px 20px 10px
        cursor: pointer
