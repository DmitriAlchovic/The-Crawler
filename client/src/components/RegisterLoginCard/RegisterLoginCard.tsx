import React from 'react';
import styles from './RegisterLoginCard.module.sass';

export default function RegisterLoginCard() {
  return (
    <div className={styles['login-register-container']}>
      <form>
        <label className={styles.label} htmlFor="email">
          Enter address*
          <input
            className={styles.input}
            id="email"
            placeholder="johndoe@gmail.com"
          />
        </label>
        <label className={styles.label} htmlFor="password">
          Password*
          <input
            className={styles.input}
            id="password"
            type="password"
            placeholder="Maximum 8 characters"
          />
        </label>
      </form>
      <div className={styles['login-register-btn']} role="button">
        Register Now
      </div>
    </div>
  );
}
