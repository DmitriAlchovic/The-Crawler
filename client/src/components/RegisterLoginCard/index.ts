import RegisterLoginCard from './RegisterLoginCard';

export default RegisterLoginCard;
