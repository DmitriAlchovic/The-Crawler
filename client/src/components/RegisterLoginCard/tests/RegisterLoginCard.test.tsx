import React from 'react';
import { render } from '@testing-library/react';
import * as reduxHooks from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import RegisterLoginCard from '../RegisterLoginCard';

jest.mock('react-redux');

describe('RegisterLoginCard', () => {
  it('should create empty component', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue('');
    const component = render(
      <BrowserRouter>
        <RegisterLoginCard hasDefaultLogin={false} />
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
