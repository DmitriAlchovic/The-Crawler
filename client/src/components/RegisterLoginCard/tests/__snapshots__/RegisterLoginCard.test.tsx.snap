// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterLoginCard should create empty component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-register-container"
      >
        <form>
          <label
            class="label"
            for="email"
          >
            Enter address
            <div
              class="input-container"
            >
              <img
                alt="envelope"
                src="mail.svg"
              />
              <input
                class="input"
                name="email"
                placeholder="johndoe@gmail.com"
                type="email"
                value=""
              />
            </div>
          </label>
          <label
            class="label"
            for="password"
          >
            Password
            <div
              class="input-container"
            >
              <img
                alt="key"
                src="key.svg"
              />
              <input
                class="input"
                minlength="8"
                name="password"
                placeholder="Maximum 8 characters"
                type="password"
                value=""
              />
            </div>
          </label>
        </form>
        <div
          class="input-error"
        >
          <div />
          <div />
        </div>
        <button
          class="login-register-btn"
          type="button"
        >
          Register Now
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-register-container"
    >
      <form>
        <label
          class="label"
          for="email"
        >
          Enter address
          <div
            class="input-container"
          >
            <img
              alt="envelope"
              src="mail.svg"
            />
            <input
              class="input"
              name="email"
              placeholder="johndoe@gmail.com"
              type="email"
              value=""
            />
          </div>
        </label>
        <label
          class="label"
          for="password"
        >
          Password
          <div
            class="input-container"
          >
            <img
              alt="key"
              src="key.svg"
            />
            <input
              class="input"
              minlength="8"
              name="password"
              placeholder="Maximum 8 characters"
              type="password"
              value=""
            />
          </div>
        </label>
      </form>
      <div
        class="input-error"
      >
        <div />
        <div />
      </div>
      <button
        class="login-register-btn"
        type="button"
      >
        Register Now
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Slide should create empty comment input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-register-container"
      >
        <form>
          <label
            class="label"
            for="email"
          >
            Enter address
            <div
              class="input-container"
            >
              <img
                alt="envelope"
                src="mail.svg"
              />
              <input
                class="input"
                name="email"
                placeholder="johndoe@gmail.com"
                type="email"
                value=""
              />
            </div>
          </label>
          <label
            class="label"
            for="password"
          >
            Password
            <div
              class="input-container"
            >
              <img
                alt="key"
                src="key.svg"
              />
              <input
                class="input"
                minlength="8"
                name="password"
                placeholder="Maximum 8 characters"
                type="password"
                value=""
              />
            </div>
          </label>
        </form>
        <div
          class="input-error"
        >
          <div />
          <div />
        </div>
        <button
          class="login-register-btn"
          type="button"
        >
          Register Now
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-register-container"
    >
      <form>
        <label
          class="label"
          for="email"
        >
          Enter address
          <div
            class="input-container"
          >
            <img
              alt="envelope"
              src="mail.svg"
            />
            <input
              class="input"
              name="email"
              placeholder="johndoe@gmail.com"
              type="email"
              value=""
            />
          </div>
        </label>
        <label
          class="label"
          for="password"
        >
          Password
          <div
            class="input-container"
          >
            <img
              alt="key"
              src="key.svg"
            />
            <input
              class="input"
              minlength="8"
              name="password"
              placeholder="Maximum 8 characters"
              type="password"
              value=""
            />
          </div>
        </label>
      </form>
      <div
        class="input-error"
      >
        <div />
        <div />
      </div>
      <button
        class="login-register-btn"
        type="button"
      >
        Register Now
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Slide should create empty component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-register-container"
      >
        <form>
          <label
            class="label"
            for="email"
          >
            Enter address
            <div
              class="input-container"
            >
              <img
                alt="envelope"
                src="mail.svg"
              />
              <input
                class="input"
                name="email"
                placeholder="johndoe@gmail.com"
                type="email"
                value=""
              />
            </div>
          </label>
          <label
            class="label"
            for="password"
          >
            Password
            <div
              class="input-container"
            >
              <img
                alt="key"
                src="key.svg"
              />
              <input
                class="input"
                minlength="8"
                name="password"
                placeholder="Maximum 8 characters"
                type="password"
                value=""
              />
            </div>
          </label>
        </form>
        <div
          class="input-error"
        >
          <div />
          <div />
        </div>
        <button
          class="login-register-btn"
          type="button"
        >
          Register Now
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-register-container"
    >
      <form>
        <label
          class="label"
          for="email"
        >
          Enter address
          <div
            class="input-container"
          >
            <img
              alt="envelope"
              src="mail.svg"
            />
            <input
              class="input"
              name="email"
              placeholder="johndoe@gmail.com"
              type="email"
              value=""
            />
          </div>
        </label>
        <label
          class="label"
          for="password"
        >
          Password
          <div
            class="input-container"
          >
            <img
              alt="key"
              src="key.svg"
            />
            <input
              class="input"
              minlength="8"
              name="password"
              placeholder="Maximum 8 characters"
              type="password"
              value=""
            />
          </div>
        </label>
      </form>
      <div
        class="input-error"
      >
        <div />
        <div />
      </div>
      <button
        class="login-register-btn"
        type="button"
      >
        Register Now
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
